packages/shared/src/__tests__/__examples__/exitActions.ts

```diff
@@ -0,0 +1,34 @@
+import { createMachine } from "xstate";
+
+const exitActionsMachine = createMachine(
+  {
+    initial: "a",
+    tsTypes: {} as import("./exitActions.typegen").Typegen0,
+    schema: {
+      events: {} as { type: "FOO"; foo: string },
+    },
+    states: {
+      a: {
+        exit: "doSomethingWithFoo",
+        on: {
+          FOO: {
+            target: "b",
+          },
+        },
+      },
+      b: {},
+    },
+  },
+  {
+    actions: {
+      doSomethingWithFoo: (context, event) => {
```

Comment:
I often add an extra event to declared ones and verify that it actually has not been provided here. Cause right now this would kinda pass if the computed type would be `TEvent | { type: 'xstate.stop' }`